### PR TITLE
Add create gradle project

### LINKED ISSUE #3101: Offer a Gradle project template via gradle init
https://stackoverflow.com/questions/34856834/create-custom-build-init-type-for-gradle-build-init-plugin

`gradle init --type quarkus` would be awesome

Check if that can be done so we decide to keep it in 1.0 or not

### PR DESCRIPTION
**Pull request by Dufgui**
fix #3101

### CODE REVIEW

File: devtools/common/src/main/java/io/quarkus/maven/utilities/MojoUtils.java [hunk 1]
```diff
@@ -329,4 +335,21 @@ public Attribute(String name, String value) {
         }
     }
 
+    public static String[] readGavFromSettingsGradle(ByteArrayInputStream buildFileInputStream, String[] gavIn) {
+        String[] gavOut = Arrays.copyOf(gavIn, gavIn.length);
+        try (Scanner scanner = new Scanner(buildFileInputStream)) {
+            while (scanner.hasNextLine()) {
+                String currentLine = scanner.nextLine();
+                if (currentLine.startsWith("group")) {
```
Comment (aloubyansky) [inline devtools/common/src/main/java/io/quarkus/maven/utilities/MojoUtils.java]: It looks like these property names here and in the template are not consistent, i.e. the `project.` prefix used in the template.

File: devtools/common/src/main/resources/templates/basic-rest/java/build.gradle-template.ftl [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// this block is necessary to make enforcedPlatform work for Quarkus plugin available
+// only locally (snapshot) that is also importing the Quarkus BOM
+buildscript {
+    repositories {
+        mavenLocal()
+    }
+    dependencies {
+        classpath "io.quarkus:quarkus-gradle-plugin:${quarkus_version}"
+    }
+}
```
Comment (aloubyansky) [inline devtools/common/src/main/resources/templates/basic-rest/java/build.gradle-template.ftl]: BTW, we might not need this anymore with newer versions of Gradle. It's not critical though for this specific PR.

File: devtools/common/src/main/resources/templates/basic-rest/java/build.gradle-template.ftl [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+apply plugin: 'io.quarkus'
+
+repositories {
+     mavenLocal()
+     mavenCentral()
+}
+
+dependencies {
+    implementation enforcedPlatform("io.quarkus:quarkus-bom:${quarkus_version}")
```
Comment (aloubyansky) [inline devtools/common/src/main/resources/templates/basic-rest/java/build.gradle-template.ftl]: As the next step, we could leave properties in the template and use `gradle.properties` as we do in quickstarts.

**Review by aloubyansky — COMMENTED** [review]
I tried generating a project using `mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create -DbuildTool=GRADLE -e` which, tbh, I find a bit weird, i.e. using maven to generate a gradle project. It looks like it did it except this probably just a noise error:

````
Caused by: java.nio.file.NoSuchFileException: /home/aloubyansky/playground/gradle-create-project/./pom.xml
    at sun.nio.fs.UnixException.translateToIOException (UnixException.java:86)
    at sun.nio.fs.UnixException.rethrowAsIOException (UnixException.java:102)
    at sun.nio.fs.UnixException.rethrowAsIOException (UnixException.java:107)
    at sun.nio.fs.UnixFileSystemProvider.newByteChannel (UnixFileSystemProvider.java:214)
    at java.nio.file.Files.newByteChannel (Files.java:361)
    at java.nio.file.Files.newByteChannel (Files.java:407)
    at java.nio.file.Files.readAllBytes (Files.java:3152)
    at io.quarkus.cli.commands.writer.FileProjectWriter.getContent (FileProjectWriter.java:60)
    at io.quarkus.cli.commands.AddExtensions.<init> (AddExtensions.java:32)
    at io.quarkus.maven.CreateProjectMojo.execute (CreateProjectMojo.java:162)
````

I couldn't build the project though
````
FAILURE: Build failed with an exception.

* Where:
Settings file '/home/aloubyansky/playground/gradle-create-project/settings.gradle' line: 16

* What went wrong:
A problem occurred evaluating settings 'gradle-create-project'.
> Could not get unknown property 'project' for settings 'gradle-create-project' of type org.gradle.initialization.DefaultSettings.
````

File: devtools/common/src/main/resources/templates/basic-rest/java/build.gradle-template.ftl (hunk 2 above)
Comment (Dufgui) [reply devtools/common/src/main/resources/templates/basic-rest/java/build.gradle-template.ftl]: ok

**Comment by Dufgui** [comment]
I miss AddExtensions implementation. So it's fixed now. So you can review and test it again.
I think we can link all gradle task (add extension and list extension to new created command). But maybe in another ticket

**Comment by Dufgui** [comment]
many thanks @aloubyansky

**Review by aloubyansky — APPROVED** [review]
It's looking pretty good to me. Further improvements can be added by creating their own issue. Thanks!

**Pull request merged**